### PR TITLE
feat: add issues in URL and handle history state

### LINKED ISSUE #118: Deep Links
- We should push and pop browser history when opening issues. 
- We need to account for proposals and for tasks. 
- We need to automatically open the task/proposal when the page is loaded with the url representing it. Maybe a # in the URL with the issue ID will be sufficient.

### PR DESCRIPTION
**Pull request by zugdev**
Resolves #118

### CODE REVIEW

File: src/home/rendering/render-github-issues.ts [hunk 1]
```diff
@@ -149,6 +149,40 @@ export function viewIssueDetails(full: GitHubIssue) {
   modal.classList.add("active");
   modal.classList.remove("error");
   document.body.classList.add("preview-active");
+
+  updateURLWithIssueID(full.id);
```
Comment (0x4007) [inline src/home/rendering/render-github-issues.ts]: Strict pascal case lint shouldnt let this work

**Review by 0x4007 — COMMENTED** [review]
Try running linter. Commit hooks shouldnt have let you commit these

**Comment by 0x4007** [comment]
Just tested. Looks like it is almost perfect. I think ideally the app mode should change to proposals when viewing a proposal though. 

https://4a76efd5.devpool-directory-ui.pages.dev/?issue=2499194812

This is a proposal but when I X out of the modal, its in the tasks view. Any good solution for this?

**Comment by zugdev** [comment]
Yeah that's easy will check `isProposalsViewer`, should be effective enough.

**Comment by zugdev** [comment]
Should be good to go now. I have added a proposal parameter to the URL if we are in proposal view. Also it changes the toggle from Directory to Proposals based on URL.

**Review by 0x4007 — APPROVED** [review]

**Pull request merged**